### PR TITLE
Fix S1144 FP: Getters/Setters of property with attribute are being flagged

### LINKED ISSUE #9219: Fix S1144 FP: Getters/Setters of property with attribute are being flagged
### Description

Members with an attribute are being flagged. The documentation states they should be exceptions to the rule.
image

### Repro steps

image

### Expected behavior
The setter should not be flagged.

### Actual behavior
The setter is flagged.

### Related information
SonarAnalyzer.CSharp: 9.24.0.89429
Visual Studio: 17.9.6
Dotnet version: .NET 8
Operating System: Windows 10 Enterprise, 21H2

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Fixes #9219

Reviewing per-commit might be helpful.
Definitely check `ignore whitespace`, since file-scoped namespace destroys reviewability.

Commit 1: The actual change
Commit 2: UTs
Commit 3: CaYC

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs [hunk 1]
```diff
@@ -20,543 +20,550 @@
...
+
+        private static bool IsRemovable(ISymbol symbol) =>
+            symbol is { IsImplicitlyDeclared: false, IsVirtual: false }
+            && !HasAttributes(symbol)
+            && !symbol.IsSerializableMember()
+            && !symbol.ContainingType.IsInterface()
+            && symbol.GetInterfaceMember() is null
+            && symbol.GetOverriddenMember() is null;
+
+        private static bool HasAttributes(ISymbol symbol)
```
Comment (gregory-paidis-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs:534]: This is the important part.
I check for the getter/setter and the property itself for attributes in the special case.

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM. Left two small comments.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs [hunk 2]
```diff
@@ -531,8 +531,16 @@ private static bool IsDeclaredInPartialClass(ISymbol methodSymbol)
                 && symbol.GetInterfaceMember() == null
                 && symbol.GetOverriddenMember() == null;
 
-            private static bool HasAttributes(ISymbol symbol) =>
-                symbol.GetAttributes().Any(x => !x.AttributeClass.Is(KnownType.System_NonSerializedAttribute));
+            private static bool HasAttributes(ISymbol symbol)
+            {
+                IEnumerable<AttributeData> attributes = symbol.GetAttributes();
+                if (symbol is IMethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet} method
```
[the remark this answers is not in the document]
Comment (gregory-paidis-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs]: Can you give me an example?
Event handlers add/removal is a delegate, it doesn't have a specific "body", as in `get;` and `set;`, so it cannot be "private". The way you add things is by `+=`, which is syntactic sugar over .NET handling multicast delegates.

If I am missing something, please give me a snippet that should not raise after the change you propose.

[the remark this answers is not in the document]
Comment (zsolt-kolbay-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs]: My bad, I forgot add/remove cannot be private.

**Pull request merged**